SpecsUtils/src/pt/up/fe/specs/util/SpecsAsm.java

```diff
@@ -148,42 +198,86 @@ public static int mbCompareUnsigned(int input1, int input2) {
         return SpecsBits.clearBit(31, result);
     }
 
+    /**
+     * Performs a logical left shift on a 32-bit integer.
+     *
+     * @param input1 the operand to shift
+     * @param input2 the number of positions to shift
+     * @return the result of the shift
+     */
     public static int shiftLeftLogical(int input1, int input2) {
         return input1 << input2;
     }
 
+    /**
+     * Performs an arithmetic right shift on a 32-bit integer.
+     *
+     * @param input1 the operand to shift
+     * @param input2 the number of positions to shift
+     * @return the result of the shift
+     */
     public static int shiftRightArithmetical(int input1, int input2) {
         return input1 >> input2;
     }
 
+    /**
+     * Performs a logical right shift on a 32-bit integer.
+     *
+     * @param input1 the operand to shift
+     * @param input2 the number of positions to shift
+     * @return the result of the shift
+     */
     public static int shiftRightLogical(int input1, int input2) {
         return input1 >>> input2;
     }
 
     /**
+     * Performs a logical left shift on a 32-bit integer, taking into account a mask.
      *
-     * @param input1
-     * @param input2
-     * @param input3
-     *            the number of LSB bits of input2 to take into account
-     * @return
+     * @param input1 the operand to shift
+     * @param input2 the number of positions to shift
+     * @param input3 the number of LSB bits of input2 to take into account
+     * @return the result of the shift
      */
     public static int shiftLeftLogical(int input1, int input2, int input3) {
         input2 = SpecsBits.mask(input2, input3);
         return input1 << input2;
     }
 
+    /**
+     * Performs an arithmetic right shift on a 32-bit integer, taking into account a mask.
+     *
+     * @param input1 the operand to shift
+     * @param input2 the number of positions to shift
+     * @param input3 the number of LSB bits of input2 to take into account
+     * @return the result of the shift
+     */
     public static int shiftRightArithmetical(int input1, int input2, int input3) {
         input2 = SpecsBits.mask(input2, input3);
         return input1 >> input2;
     }
 
+    /**
+     * Performs a logical right shift on a 32-bit integer, taking into account a mask.
+     *
+     * @param input1 the operand to shift
+     * @param input2 the number of positions to shift
+     * @param input3 the number of LSB bits of input2 to take into account
+     * @return the result of the shift
+     */
     public static int shiftRightLogical(int input1, int input2, int input3) {
         input2 = SpecsBits.mask(input2, input3);
         return input1 >>> input2;
     }
 
+    /**
+     * Neutral carry value for addition operations.
+     */
     public static final int CARRY_NEUTRAL_ADD = 0;
+
+    /**
+     * Neutral carry value for subtraction operations.
+     */
     public static final int CARRY_NEUTRAL_SUB = 1;
```

Comment:
- For consistency, move these static variables to the beginning of the file
- Make them private, if they are being used outside of the class, add static accessors